packages/react-components/react-field/Spec.md

```diff
@@ -2,62 +2,338 @@
+  - `'success'` - Green check icon, neutral color text
+  - `undefined` (default): No status icon, neutral color text
+- **Error**: Some control types (like `Input` and `Combobox`) have a prop that makes the border red. This prop will be set `status="error"`.
+
+Field also forwards some props from the wrapped component to the label as well:
+
+- **Size**: If the wrapped component supports a `size` prop, it will also be applied to the field's label.
+- **Required**: If set, the Label will get a required asterisk: `*`
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+### FieldComponent
+
+The `FieldComponent` type defines the minimum set of props that the wrapped component must support. This is used for the generic types as the requirement for the type parameter: `FieldProps<T extends FieldComponent>`
+
+```ts
+/**
+ * The minimum requirement for a component used by Field.
+ *
+ * Note: the use of VoidFunctionComponent means that component is not *required* to have a children prop,
+ * but it is still allowed to have a children prop.
+ */
+export type FieldComponent = React.VoidFunctionComponent<
+  Pick<
+    React.HTMLAttributes<HTMLElement>,
+    'id' | 'className' | 'style' | 'aria-labelledby' | 'aria-describedby' | 'aria-invalid' | 'aria-errormessage'
+  >
+>;
+```
+
+### Slots
+
+_Note: TypeScript crashes if the `Slot` type is used with a template type parameter. The `SlotComponent` type is a simplified version of that type, which only supports `React.ComponentType`/`React.VoidFunctionComponent`._
+
+```ts
+export type FieldSlots<T extends FieldComponent> = {
+  root: NonNullable<Slot<'div'>>;
+
+  /**
+   * The underlying component wrapped by this field.
+   *
+   * This is the PRIMARY slot: all intrinsic HTML properties will be applied to this slot,
+   * except `className` and `style`, which remain on the root slot.
+   */
+  fieldComponent: SlotComponent<T>;
+
+  /**
+   * The label associated with the field.
+   */
+  label?: SlotComponent<typeof Label>;
+
+  /**
+   * A status or validation message. The appearance of the statusText depends on the value of the `status` prop.
+   */
+  statusText?: Slot<'span'>;
+
+  /**
+   * The icon associated with the status. If the `status` prop is set, this will default to a corresponding icon.
+   *
+   * This will only be displayed if `statusText` is set.
+   */
+  statusIcon?: Slot<'span'>;
+
+  /**
+   * Additional text below the field.
+   */
+  helperText?: Slot<'span'>;
+};
+```
+
+### Props
+
+```ts
+export type FieldProps<T extends FieldComponent> = ComponentProps<Partial<FieldSlots<T>>, 'fieldComponent'> & {
+  /**
+   * The orientation of the label relative to the field component.
+   * This only affects the label, and not the statusText or helperText (which always appear below the field component).
+   *
+   * @default vertical
+   */
+  fieldOrientation?: 'vertical' | 'horizontal';
```

Comment:
I named this `fieldOrientation` to avoid conflicts with other components' props. `orientation` conflicts with the Slider prop, and `labelPosition` conflicts with the Checkbox prop. The latter also implies that you need a label for this to have an effect, but it still modifies the layout regardless of whether there's a label.

Any other ideas on a better name are welcome!